datasets/my/aob_sanctions/crawler.py

```diff
@@ -0,0 +1,93 @@
+from typing import Generator, Dict, cast
+from lxml.etree import _Element
+from normality import collapse_spaces, slugify
+
+from zavod import Context, helpers as h
+
+
+def parse_table(table: _Element) -> Generator[Dict[str, str], None, None]:
+    """
+    First we find the headers by searching the th tags, then we iterate over the tr tags (i.e. rows).
+
+    Returns:
+        A generator that yields a dictionary of the table columns and values. The keys are the
+        column names and the values are the column values.
+    Raises:
+        AssertionError: If the headers don't match what we expect.
+    """
+    headers = [th.text_content().strip() for th in table.findall(".//th")]
+    for row in table.findall(".//tr")[1:]:
+        if headers is None:
+            headers = []
+            for el in row.findall("./td"):
+                headers.append(slugify(el.text_content()))
+            continue
+
+        cells = []
+        for el in row.findall("./td"):
+            cells.append(collapse_spaces(el.text_content()))
+
+        # There are some incorrect rows with a smaller
+        # number of values, we'll just ignore those
+        if len(cells) != len(headers):
+            continue
+
+        # The table has a last row with all empty values
+        if all(c == "" for c in cells):
+            continue
+
+        yield {hdr: c for hdr, c in zip(headers, cells)}
+
+
+def crawl_item(input_dict: dict, context: Context):
+    name_col = "Auditor" if "Auditor" in input_dict else "Parties Involved"
+    date_col = (
+        "Date of Action" if "Date of Action" in input_dict else "Date of AOB's Action"
+    )
+    description_col = (
+        "Brief Description of Breach"
+        if "Brief Description of Breach" in input_dict
+        else "Brief Description of Misconduct"
+    )
+    reason_col = (
+        "Nature of Breach"
+        if "Nature of Breach" in input_dict
+        else "Nature of Misconduct"
+    )
+    name = input_dict.pop(name_col)
+
+    clean_names = context.lookup("clean_names", name)
+    clean_names = cast("List[str]", clean_names.names)
```

Comment:
I see all your effort with the name lookup, but in this case, how wrong would `clean_name = h.multi_split(name, ["(", " of "])[0]` be?

```suggestion
    res = context.lookup("clean_names", name)
    clean_names = cast("List[str]", res.names)
```

our naming convention is `res`, as in `result`, I think. Not super clear, but at least not changing the type of a variable

Reply:
I haven't noticed this pattern when I was doing it, indeed most of them fall into this categories. Do you want me to update it to use that? There are some instances where this will fail, but we could use a lookup for those, for example:

```
Mohd. Neezal bin Md. Noordin Partner of AljeffriDean who was the engagement partner in the audit of a public interest entity (PIE) for the financial year ended 31 March 2011.
```

This would return with the `Partner` in the name, but it is not correct.